content/docs/guides/dockerswarm.md

```diff
@@ -0,0 +1,227 @@
+ingested by a Prometheus server.
+
+You can enable them by editing `/etc/docker/daemon.json` and setting the
+following properties:
+
+```json
+{
+  "metrics-addr" : "0.0.0.0:9323",
+  "experimental" : true
+}
+```
+
+Instead of `0.0.0.0`, you can set the IP of the Docker Swarm node.
+
+A restart of the daemon is required to take the new configuration into account.
+
+The [Docker documentation][dockermetrics] contains more info about this.
+
+Then, you can configure Prometheus to scrape the Docker daemon, by providing the
+following `prometheus.yml` file:
+
+
+```yaml
+scrape_configs:
+  # Make Prometheus scrape itself for metrics.
+  - job_name: 'prometheus'
+    static_configs:
+    - targets: ['localhost:9090']
+
+  # Create a job for Docker daemons.
+  - job_name: 'docker'
+    dockerswarm_sd_configs:
+      - host: unix:///var/run/docker.sock
+        role: nodes
+    relabel_configs:
+      # Fetch metrics on port 9323.
+      - source_labels: [__meta_dockerswarm_node_address]
+        target_label: __address__
+        replacement: $1:9323
+      # Set hostname as instance label
+      - source_labels: [__meta_dockerswarm_node_hostname]
+        target_label: instance
+```
+
+**Note**: for the nodes role, you can also use the `port` parameter of
+`dockerswarm_sd_configs`. However, using `relabel_configs` is recommended as it
+enables Prometheus to reuse the same API calls across identical Docker Swarm
+configurations.
+
+## Monitoring Containers
+
+Let's now deploy a service in our Swarm. We will deploy [cadvisor][cad], which
+exposes container resources metrics:
+
+```shell
+docker service create --name cadvisor -l prometheus-job=cadvisor \
+    --mode=global --publish target=8080,mode=host \
+    --mount type=bind,src=/var/run/docker.sock,dst=/var/run/docker.sock,ro \
+    --mount type=bind,src=/,dst=/rootfs,ro \
+    --mount type=bind,src=/var/run,dst=/var/run \
+    --mount type=bind,src=/sys,dst=/sys,ro \
+    --mount type=bind,src=/var/lib/docker,dst=/var/lib/docker,ro \
+    google/cadvisor -docker_only
+```
+
+This is a minimal `prometheus.yml` file to monitor it:
+
+```yaml
+scrape_configs:
+  # Make Prometheus scrape itself for metrics.
+  - job_name: 'prometheus'
+    static_configs:
+    - targets: ['localhost:9090']
+
+  # Create a job for Docker Swarm containers.
+  - job_name: 'dockerswarm'
+    dockerswarm_sd_configs:
+      - host: unix:///var/run/docker.sock
+        role: tasks
+    relabel_configs:
+      # Only keep containers that should be running.
+      - source_labels: [__meta_dockerswarm_task_desired_state]
+        regex: running
+        action: keep
+      # Only keep containers that have a `prometheus-job` label.
+      - source_labels: [__meta_dockerswarm_service_label_prometheus_job]
+        regex: .+
+        action: keep
+      # Use the task labels that are prefixed by `prometheus-`.
+      - regex: __meta_dockerswarm_service_label_prometheus_(.+)
+        action: labelmap
```

Comment:
Let's be careful here, having labelmap in k8 examples has lead to excessive target labels by many users - which then caused the usual problems from that.

Reply:
Fixed